journal/src/main/java/io/camunda/zeebe/journal/file/Segment.java

```diff
@@ -240,7 +260,6 @@ private void markForDeletion() {
       return;
     }
 
-    writer.close();
```

Comment:
The writer close wasn't really doing anything but flushing. But since we closed it only on deletion, it seemed silly to flush right before deleting :smile:

Reply:
:question: We also close segments while shutting down, right? Shouldn't we flush before shutting down? But this can be handled somewhere higher up in the journal.

Reply:
As far as I could tell, we only closed the writer when deleting the segment, not when closing it. So we weren't flushing either on closing.

The `SegmentedJournalWriter#close` method was calling the `SegmentWriter#close` as well, but we also never closed the `SegmentedJournalWriter`. Was that an over sight?

I also agree we might want to flush before closing, but as you said we can probably do this on journal close.